crates/rpc-server/src/registry.rs

```diff
@@ -479,9 +501,9 @@ impl RequestSubmitter {
                 );
                 for entry in items {
                     let maybe_ok = match entry.item.clone() {
-                        FeeItem::Tx(tx) => mem_pool.push_transaction(tx),
+                        FeeItem::Tx(tx) => smol::block_on(mem_pool.push_transaction(tx)),
```

Comment:
.await

Reply:
```
    = help: within `MemStateTree<'_>`, the trait `Sync` is not implemented for `RefCell<HashSet<gw_common::H256>>`
    = note: required because it appears within the type `std::option::Option<RefCell<HashSet<gw_common::H256>>>`
    = note: required because it appears within the type `gw_store::state::state_tracker::StateTracker`
    = note: required because it appears within the type `MemStateTree<'_>`
    = note: required because of the requirements on the impl of `std::marker::Send` for `&MemStateTree<'_>`
```

If we call `.await` we need to implement Send on the MemStateTree, same for the other comment.
IMO we can figure out this solution in another PR.

Reply:
Ok